src/_pytest/fixtures.py

```diff
@@ -758,7 +758,7 @@ def __init__(
         self,
         request: "FixtureRequest",
         scope: "_Scope",
-        param,
+        param: Optional[Any],
```

Comment:
I don't think `Optional` is correct here, there is no `None` here (unless it's the actual param value).

One other complication is that `param` can be `NOTSET` and then `self.param` is only set conditionally. There is currently no way to represent this in Python (happily). This is the reason it isn't annotated.

But overall I'm fine with `request.param` just being `Any` for now, until we can clean this up. So if you remove the `Optional` I'd be OK.